.github/workflows/testing-all-oses.yml

```diff
@@ -0,0 +1,53 @@
+name: Test MSS
+
+on:
+  push:
+    branches:
+      - develop
+      - stable
+      - 'GSOC**'
+  pull_request:
+    branches:
+      - develop
+      - stable
+      - 'GSOC**'
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["macos-14", "ubuntu-latest"]
+        order: ["normal", "reverse"]
+    steps:
+    - uses: actions/checkout@v4
+    - name: Build requirements.txt file
+      run: |
+        sed -n '/^requirements:/,/^test:/p' localbuild/meta.yaml |
+          sed -e "s/.*- //" |
+          sed -e "s/menuinst.*//" |
+          sed -e "s/.*://" > requirements.tmp.txt
+        cat requirements.d/development.txt >> requirements.tmp.txt
+        sed -e '/^$/d' -e '/^#.*$/d' requirements.tmp.txt > requirements.txt
+        rm requirements.tmp.txt
+        cat requirements.txt
+    - name: Get current year and calendar week
+      id: year-and-week
+      run: echo "year-and-week=$(date +%Y-%V)" >> "$GITHUB_OUTPUT"
+    - uses: mamba-org/setup-micromamba@v1
+      with:
+        environment-file: requirements.txt
+        environment-name: ci
+        cache-environment: true
+        # Set the cache key in a way that the cache is invalidated every week on monday
+        cache-environment-key: environment-${{ steps.year-and-week.outputs.year-and-week }}
+    - name: Run tests
+      timeout-minutes: 10
+      # The ignored files can somehow cause the test suite to timeout.
+      # I have no idea yet on why this happens and how to fix it.
+      # Even a module level skip is not enough, they need to be completely ignored.
+      # TODO: fix those tests and drop the ignores
+      run: micromamba run -n ci env QT_QPA_PLATFORM=offscreen pytest -v -n logical --durations=20 --cov=mslib
+        --ignore=tests/_test_msui/test_sideview.py --ignore=tests/_test_msui/test_topview.py --ignore=tests/_test_msui/test_wms_control.py
```

Comment:
is this 20 a total time? and what happens when the tests timeout of 10 is reached?

Reply:
While debugging a test I recently had seen that it sometimes gets into the http_auth dialog but that is not defined. It is not reproducable, it then enters the password there and fails ;)

Reply:
The 20 just means "print the duration of the 20 slowest tests". I copied that over from the other testing workflow. If the 10 minute step timeout is reached the workflow is aborted and marked as failed.

Yes, there are still some flaky tests left. The files that are disabled here seem to cause at least some of that, since I have seen the same symptoms that they exhibit on macOS also happen on Linux, just much less frequently which is why it isn't much of a problem in the other workflow.

Reply:
Note: I've increased the step timeout because the macOS 13 runners are a bit slower than the macOS 14 ones.